internal/pkg/manifest/svc.go

```diff
@@ -39,7 +49,112 @@ type Service struct {
 
 // ServiceImage represents the service's container image.
 type ServiceImage struct {
-	Build *string `yaml:"build"` // Path to the Dockerfile.
+	Build BuildArgsOrString `yaml:"build"` // Path to the Dockerfile.
+}
+
+// BuildConfig populates a docker.BuildArguments struct from the fields available in the manifest.
+// Prefer the following hierarchy:
+// 1. Specific dockerfile, specific context
+// 2. Specific dockerfile, context = dockerfile dir
+// 3. "Dockerfile" located in context dir
+// 4. "Dockerfile" located in ws root.
+func (s *ServiceImage) BuildConfig(rootDirectory string) *docker.BuildArguments {
+	df := s.dockerfile()
+	ctx := s.context()
+	if df != "" && ctx != "" {
+		df = filepath.Join(rootDirectory, df)
+		ctx = filepath.Join(rootDirectory, ctx)
+	} else if df != "" && ctx == "" {
+		df = filepath.Join(rootDirectory, df)
+		ctx = filepath.Dir(df)
+	} else if df == "" && ctx != "" {
+		ctx = filepath.Join(rootDirectory, ctx)
+		df = filepath.Join(ctx, dockerfileDefaultName)
+	} else {
+		ctx = rootDirectory
+		df = filepath.Join(rootDirectory, dockerfileDefaultName)
+	}
```

Comment:
nit: prefer replacing if-else statements with guard-clauses https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html
```suggestion
	if df != "" && ctx != "" {
		return &docker.BuildArguments {
			Dockerfile: filepath.Join(rootDirectory, df),
			Context:    filepath.Join(rootDirectory, ctx),
			Args: s.args(),
		}
	} 
	if df != "" && ctx == "" {
		return &docker.BuildArguments {
			Dockerfile: filepath.Join(rootDirectory, df),
			Context:    filepath.Dir(df),
			Args: s.args(),
		}
	} 
	if df == "" && ctx != "" {
		return &docker.BuildArguments {
			Dockerfile: filepath.Join(ctx, dockerfileDefaultName),
			Context:    filepath.Join(rootDirectory, ctx),
			Args: s.args(),
		}
	} 
	return &docker.BuildArguments {
		Dockerfile: filepath.Join(rootDirectory, dockerfileDefaultName),
		Context:    rootDirectory,
		Args: s.args(),
	}
```